src/memory.rs

```diff
@@ -0,0 +1,63 @@
+use std::cell::Cell;
+use rand::rngs::StdRng;
+
+use rustc_mir::interpret::{Pointer, Allocation, AllocationExtra, InterpResult};
+use rustc_target::abi::Size;
+
+use crate::{stacked_borrows, intptrcast};
+use crate::stacked_borrows::{Tag, AccessKind};
+
+#[derive(Default, Clone, Debug)]
+pub struct MemoryState {
+    pub stacked: stacked_borrows::MemoryState,
+    pub intptrcast: intptrcast::MemoryState,
```

Comment:
I think a more consistent field name would be `stacked_borrows` -- let's consistently use the module name.